lib/animina/accounts/resources/photo.ex

```diff
@@ -175,4 +195,75 @@ defmodule Animina.Accounts.Photo do
       reference :user, on_delete: :delete
     end
   end
+
+  def create_optimized_photos(record) do
+    case check_if_image_magick_is_installed() do
+      {:ok, _} ->
+        resize_images_with_imagemagick(record)
+
+      {:error, _} ->
+        copy_images_directly(record)
+    end
+  end
+
+  defp copy_images_directly(record) do
+    for type <- [:thumbnail, :normal, :big] do
+      OptimizedPhoto.create(%{
+        image_url: record.filename,
+        type: type,
+        user_id: record.user_id,
+        photo_id: record.id
+      })
+    end
+  end
+
+  # TODO: Confirm if the dimensions are  okay for each type
+  defp resize_images_with_imagemagick(record) do
```

Comment:
Does it make sense to give a width and a height? We don't know if these are squares. Probably they are not. Just set the width and leave the height to be figured out automatically.

Reply:
We can use this instead 
`resize_to_fill(image, params)`
Resize the image to fit within the specified dimensions while retaining the aspect ratio of the original image.
If necessary, crop the image in the larger dimension.


SInce you have imagemagick , can you confirm if we can do 
```
 |> Mogrify.resize("#{width}")
```

Reply:
Assume that it works. I'll test it once everything is put together. In case there is a bug in the Mogrify part I'll fix it.